imapclient/imapclient.py

```diff
@@ -383,6 +383,18 @@ def plain_login(self, identity, password, authorization_identity=None):
         except exceptions.IMAPClientError as e:
             raise exceptions.LoginError(str(e))
 
+    def sasl_login(self, mech_name, mech_callable):
+        """Authenticate using a provided SASL mechanism (requires server support).
+
+        The 'mech_callable' must accept a parameter with the server challenge
+        and return the client response. It will be called as many times as the
+        server produces challenges.
+        """
```

Comment:
Could you add a small usage example in the docstring? It would help to understand how `mech_callable` should be built. Thank you.